src/KeyVault/KeyVault/SecurityDomain/Models/SecurityDomainClient.cs

```diff
@@ -59,67 +64,178 @@ public string DownloadSecurityDomain(string hsmName, IEnumerable<X509Certificate
+            }
+        }
 
-            PrepareRequest(httpRequest);
+        /// <summary>
+        /// Create security domain HTTP request
+        /// </summary>
+        /// <param name="method"></param>
+        /// <param name="hsmName"></param>
+        /// <param name="path">e.g. /securitydomain/download</param>
+        /// <param name="content">optional request body</param>
+        /// <returns></returns>
+        private HttpRequestMessage CreateRequest(HttpMethod method, string hsmName, string path, HttpContent content = null)
+        {
+            var uri = new UriBuilder(_uriHelper.CreateManagedHsmUri(hsmName))
+            {
+                Path = path,
+                Query = $"api-version={_apiVersion}"
+            }.Uri;
 
-            var httpResponseMessage = HttpClient.SendAsync(httpRequest).ConfigureAwait(false).GetAwaiter().GetResult();
+            var request = new HttpRequestMessage(method, uri)
+            {
+                Content = content
+            };
 
-            if (httpResponseMessage.IsSuccessStatusCode)
+            // add content-type header
+            if (request.Content != null)
             {
-                string response = httpResponseMessage.Content.ReadAsStringAsync().ConfigureAwait(false).GetAwaiter().GetResult();
-                var securityDomainWrapper = JsonConvert.DeserializeObject<SecurityDomainWrapper>(response);
-                ValidateDownloadSecurityDomainResponse(securityDomainWrapper);
-                return securityDomainWrapper.value;
+                request.Content.Headers.ContentType = MediaTypeHeaderValue.Parse("application/json; charset=utf-8");
             }
-            else
+
+            // add authorization header
+            try
             {
-                string response = httpResponseMessage.Content.ReadAsStringAsync().ConfigureAwait(false).GetAwaiter().GetResult();
-                _writeDebug($"Invalid security domain response: {response}");
-                throw new Exception(Resources.DownloadSecurityDomainFail);
+                var token = _credentials.GetAccessToken();
+                token.AuthorizeRequest((tokenType, tokenValue) =>
+                {
+                    request.Headers.Authorization = new AuthenticationHeaderValue(tokenType, tokenValue);
+                });
             }
-        }
-
-        private void ValidateDownloadSecurityDomainResponse(SecurityDomainWrapper securityDomainWrapper)
-        {
-            if (string.IsNullOrEmpty(securityDomainWrapper.value) || !ValidateSecurityDomainData(securityDomainWrapper.value))
+            catch (Exception ex)
             {
-                _writeDebug($"Invalid security domain response: {securityDomainWrapper.value}");
-                throw new Exception(Resources.DownloadSecurityDomainFail);
+                throw new AzPSException(Resources.InvalidSubscriptionState, ErrorKind.UserError, ex);
```

Comment:
ErrorKind.UserError doesn't match all cases

Reply:
Makes sense. I'll change it to internal error